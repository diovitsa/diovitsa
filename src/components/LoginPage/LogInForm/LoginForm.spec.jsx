import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LogInForm';

describe('LogInForm', () => {
  let component;
  let onSignIn;

  function getComponent() {
    const testingComponent = shallow(<LoginForm onSignIn={onSignIn}/>).dive();
    return testingComponent.instance();
  }

  beforeEach(() => {
    onSignIn = jest.fn();
    component = getComponent();
  });

  it('should render properly', () => {
    expect(getComponent()).toMatchSnapshot();
  });

  describe('componentDidMount', () => {

    beforeEach(() => {
      jest.spyOn(document, 'addEventListener');
      component.componentDidMount();
    });

    it('should init keydown listener', () => {
      expect(document.addEventListener).toHaveBeenCalledWith('keydown', component.keydown, false);
    });

  });

  describe('componentWillUnmount', () => {

    beforeEach(() => {
      jest.spyOn(document, 'removeEventListener');
      component.componentWillUnmount();
    });

    it('should remove keydown listener', () => {
      expect(document.removeEventListener).toHaveBeenCalledWith('keydown', component.keydown, false);
    });

  });

  describe('keydown', () => {

    beforeEach(() => {
      jest.spyOn(component, 'signIn');
    });

    describe('when email or password input is on focus', () => {
      beforeEach(() => {
        const event = {
          key: 'Enter',
          target: 'passwordInput'
        };
        component.passwordInput = 'passwordInput';
        component.keydown(event);
      });

      it('should call signIn method', () => {
        expect(component.signIn).toHaveBeenCalled();
      });
    });

    describe('when focus is not on form inputs', () => {
      beforeEach(() => {
        const event = {
          key: 'Enter',
        };

        component.passwordInput = 'passwordInput';
        component.keydown(event);
      });

      it('should call signIn method', () => {
        expect(component.signIn).not.toHaveBeenCalled();
      });
    });

    describe('when key it\'s not Enter', () => {
      beforeEach(() => {
        const event = {
          key: 'Space',
          target: 'passwordInput'
        };

        component.passwordInput = 'passwordInput';
        component.keydown(event);
      });

      it('should call signIn method', () => {
        expect(component.signIn).not.toHaveBeenCalled();
      });
    });
  });

  describe('signIn', () => {

    beforeEach(() => {
      jest.spyOn(component, 'clearInputValues');
      component.state = { email: 'testEmail', password: 'testPassword' };
      component.signIn();
    });

    it('should clear all input values', () => {
      expect(component.clearInputValues).toHaveBeenCalled();
    });

    it('should call component.props.onSignIn', () => {
      expect(component.props.onSignIn).toHaveBeenCalledWith('testEmail', 'testPassword');
    });
  });

  describe('clearInputValues', () => {

    beforeEach(() => {
      component.emailInput = { value: 'email@gmail.com' };
      component.passwordInput = { value: 'password' };

      component.clearInputValues();
    });

    it('should clear all input values', () => {
      expect(component.emailInput.value).toEqual('');
      expect(component.passwordInput.value).toEqual('');
    });

  });

});
