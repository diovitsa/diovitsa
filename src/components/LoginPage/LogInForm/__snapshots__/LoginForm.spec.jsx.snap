// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm should render properly 1`] = `
LogInForm {
  "context": Object {},
  "keydown": [Function],
  "props": Object {
    "classes": Object {
      "root": "LogInForm-root-1",
    },
    "onSignIn": [MockFunction],
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "email": "",
    "password": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <LogInForm
        classes={
          Object {
            "root": "LogInForm-root-1",
          }
        }
        onSignIn={[MockFunction]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="LogInForm-root-1"
      >
        <TextField
          label="Email"
          onChange={[Function]}
          required={false}
          select={false}
          type="email"
          variant="standard"
        />
        <TextField
          label="Password"
          onChange={[Function]}
          required={false}
          select={false}
          type="password"
          variant="standard"
        />
        <WithStyles(WithStyles(Button))
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
           Sign In
        </WithStyles(WithStyles(Button))>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;
